entity_gym/entity_gym/environment.py

```diff
@@ -184,7 +203,67 @@ class ObsBatch:
     end_of_episode_info: Dict[int, EpisodeStats]
 
 
-def batch_obs(obs: List[Observation]) -> ObsBatch:
+def merge_obs(a: Optional[ObsBatch], b: ObsBatch) -> ObsBatch:
+    """
+    merges two obs_batches
+    """
+
+    if a == None:
+        return b
+
+    assert isinstance(a, ObsBatch)
+
+    assert (
+        a.entities.keys() == b.entities.keys()
+    ), "Malformed batches. batches must contain the same entity keys to be merged"
+    assert (
+        a.action_masks.keys() == b.action_masks.keys()
+    ), "Malformed batches. batches must contain the same action mask keys to be merged"
+
+    entities = {}
+    ids: Sequence[Sequence[EntityID]] = []
+    action_masks: Dict[
+        str, Union[CategoricalActionMaskBatch, SelectEntityActionMaskBatch]
```

Comment:
We could add a type alias `ActionMaskBatch = Union[CategoricalActionMaskBatch, SelectEntityActionMaskBatch]`